includes/internal/emails/generators/class-email-generators-abstract.php

```diff
@@ -80,6 +80,30 @@ public function is_email_active() {
 		return $email && 'publish' === $email->post_status;
 	}
 
+	/**
+	 * Add action if email is active.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @internal
+	 *
+	 * @param string   $action        Action name.
+	 * @param callable $callback      Callback.
+	 * @param int      $priority      Priority.
+	 * @param int      $accepted_args Accepted arguments.
+	 */
+	protected function maybe_add_action( $action, $callback, $priority = 10, $accepted_args = 1 ) {
+		add_action(
+			$action,
+			function () use ( $action, $callback, $priority, $accepted_args ) {
```

Comment:
Is it a standard practice to use anonymous functions for deferred hooks?
My concern is about removing the action. Not sure if it is important here.

Reply:
It's not a standard practice and we should avoid using them for all actions and filters.